lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -164,6 +164,24 @@ void circt::firrtl::emitConnect(ImplicitLocOpBuilder &builder, Value dst,
     builder.create<ConnectOp>(dst, src);
 }
 
+ForeignOutputOp circt::firrtl::getForeignOutputOf(Value value) {
+  ForeignOutputOp result;
+  for (auto &use : value.getUses()) {
+    auto match = dyn_cast<ForeignOutputOp>(use.getOwner());
+    // any non-output uses cancel this.
+    if (!match)
+      return {};
+    // Must be the dest, not the src
+    if (use.getOperandNumber() != 0)
+      return {};
+    // Multiple outputs
+    if (result)
```

Comment:
If this is used with a wire-like storage, if there is an operation using this as a source AND a destination this'll return `{}`.

And just wondering if this needs to check for multiple -- depending on where things land re:verifier?

Reply:
There can't be a wire-like storage using this by construction.

Reply:
Okay, thanks.